extensions/amp-a4a/0.1/a4a-variable-source.js

```diff
@@ -113,10 +112,66 @@ export class A4AVariableSource extends VariableSource {
       return getNavigationData(this.win_, 'redirectCount');
     });
 
+    this.set('HTML_ATTR',
+        /** @type {function(...*)} */(this.htmlAttrBinding_.bind(this)));
+
     for (let v = 0; v < WHITELISTED_VARIABLES.length; v++) {
       const varName = WHITELISTED_VARIABLES[v];
       const resolvers = this.globalVariableSource_.get(varName);
       this.set(varName, resolvers.sync).setAsync(varName, resolvers.async);
     }
   }
+
+  /**
+   * Provides a binding for getting attributes from the DOM.
+   * Most such bindings are provided in src/service/url-replacements-impl, but
+   * this one needs access to this.win_.document, which if the amp-analytics
+   * tag is contained within an amp-ad tag will NOT be the parent/publisher
+   * page. Hence the need to put it here.
+   * @param {string} cssSelector Elements matching this selector will be
+   *     included, provided they have at least one of the attributeNames
+   *     set, up to a max of 10. May be URI encoded.
+   * Note: Additional params will be the names of the attributes whose values
+   * will be returned. There should be at least 1.
+   * @returns {string} A stringified JSON array containing one member for each
+   *     matching element. Each member will contain the names and values of the
+   *     specified attributes, if the corresponding element has that attribute.
+   *     Note that if an element matches the cssSelected but has none of the
+   *     requested attributes, then nothing will be included in the array
+   *     for that element.
+   */
+  htmlAttrBinding_(cssSelector) {
+    const HTML_ATTR_MAX_RETURN_SIZE = 10;
+    const attributeNames = Array.prototype.slice.call(arguments, 1);
+    if (!cssSelector || !attributeNames.length) {
+      return '[]';
+    }
+    cssSelector = decodeURI(cssSelector);
+    let elements;
+    try {
+      elements = this.win_.document.querySelectorAll(cssSelector);
+    } catch (e) {
+      const TAG = 'A4AVariableSource';
+      user().warn(TAG, `Invalid selector: ${cssSelector}`);
+      return '[]';
+    }
+    const result = [];
+    for (let i = 0; i < elements.length &&
+        result.length < HTML_ATTR_MAX_RETURN_SIZE; ++i) {
+      const currentResult = {};
+      let foundAtLeastOneAttr = false;
+      for (let j = 0; j < attributeNames.length; ++j) {
```

Comment:
Optional code golf:

elements.forEach(elem => {
   if (result.length >= HTML_ATTR_MAX_RETURN_SIZE) return;
   const currentResult = {};
   attributeNames.forEach(attr => elem.hasAttribute(attr) && currentResult[attr] = elem.getAttribute(attr));
   if (Object.keys(currentResult).length) {
       result.push(currentResult);
   }
});

Reply:
Had that originally, Justin said to switch to this. https://github.com/ampproject/amphtml/pull/13712#discussion_r171727035